### DISCUSSION TITLE
brew test-bot --only-formulae fails with "cant find python" but python isn't even in the formula

### DISCUSSION THREAD

**Discussion by mhanberg** [Homebrew/discussions#5133] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
idk, its in CI
```

### Output of `brew doctor`

```shell
idk, its in CI
```

### Description of issue

I am trying to publish a new version of a formula on my tap and its giving me this weird issue it never has before.

https://github.com/elixir-tools/homebrew-tap/actions/runs/7922699119/job/21630874909#step:8:27

```
==> Using Homebrew/homebrew-test-bot 1177d3d (Merge pull request #1017 from apainintheneck/specify-that-style-should-only-run-on-formulae)
==> Using Homebrew/brew 4.2.8-102-g65102db3f (Merge pull request #16670 from EricFromCanada/BrewTestBot-URLs)
==> Testing elixir-tools/homebrew-tap 83a7ef5 (Merge 1da413d51a33ba299d5fa4e96d69793ffbda7cd7 into 9fc59fdd3db438048f8922f3ff8f90130bc95990):

==> Running FormulaeDetect#detect_formulae!
==> git -C /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/elixir-tools/homebrew-tap fetch origin +refs/heads/main
    url               https://github.com/elixir-tools/homebrew-tap/pull/22/checks
    tap origin/main 9fc59fd (next-ls: update 0.15.0 bottle.)
    HEAD              83a7ef5 (Merge 1da413d51a33ba299d5fa4e96d69793ffbda7cd7 into 9fc59fdd3db438048f8922f3ff8f90130bc95990)
    diff_start_sha1   9fc59fdd3db438048f8922f3ff8f90130bc95990
    diff_end_sha1     83a7ef579c5eecabafa21ed293a57ca3aaa49400

    testing_formulae  elixir-tools/tap/next-ls
    added_formulae    (none)
    modified_formulae elixir-tools/tap/next-ls
    deleted_formulae  (none)
    formulae_to_fetch (none)

==> Running Formulae#run!
Error: No available formula with the name "python". Did you mean ipython, bpython, wxpython, pythran, jython, cython or ptpython?
Error: Process completed with exit code 1.
```

However, my formula does not include python as a dependency, so I have no idea why it's failing on that. There is nothing for me to change.

**Comment by mike-carey** [comment]
We are experiencing the same issue attempting to install any node dependent formula.  This was working for us yesterday and started today.  This a Linux-only issue.  We do not have this issue on MacOS.  We are also running on CI so not sure if this is all linuxbrew or just Github Actions homebrew setup itself.

When attempting to install a NodeJS cli or node itself via brew:

```
==> Downloading https://ghcr.io/v2/homebrew/core/node/manifests/21.6.2
Warning: No available formula with the name "python" (dependency of node). Did you mean ipython, bpython, wxpython, pythran, jython, cython or ptpython?
==> Searching for similarly named formulae...
==> Formulae
app-engine-python
boost-python3
bpython
...

To install app-engine-python, run:
  brew install app-engine-python
```

This boils down to the `node` formula itself (or at least ours does); however, the node formula depends_on `python@3.12` so I am not sure where this is coming from.

Our workaround for the time being is to add the `--ignore-dependencies` flag when installing our clis but obviously requires you to have node installed prior.

**Reply by mike-carey** [reply]
I have also created a workflow that simply installs `node` via brew and one can see that MacOS works just fine:

https://github.com/mike-carey/test_repo/blob/troubleshoot/node-brew-dependency/.github/workflows/install-node-via-brew.yml

I purposely prevent errors because GHA will kill the other matrix job if one errors out.

**Comment by SMillerDev** [comment]
Can you reproduce this in the brew docker container?

**Reply by parabolic** [reply]
@SMillerDev I have tried running this on docker with ubuntu 22.04 and it works.
Some combination of GH actions runner 22.04.03 and setup-homebrew (Homebrew/actions/setup-homebrew@master) action is breaking it. 
We have issues with argocd-cli (`brew install argocd`), which is a single golang binary and no dependencies.
It worked fine until today.

**Reply by ZhongRuoyu** [reply]
It seems it's because the Docker image has `homebrew/core` tapped.

**Comment by ZhongRuoyu** [comment, marked as answer]
This is being tracked at Homebrew/brew#16682.

**Reply by mike-carey** [reply]
Thank you!! Confirmed this is working again with that fix.
